src/v/storage/disk_log_impl.cc

```diff
@@ -419,9 +419,9 @@ ss::future<compaction_result> disk_log_impl::compact_adjacent_segments(
     std::copy(range.first, range.second, std::back_inserter(segments));
 
     if (stlog.is_enabled(ss::log_level::debug)) {
-        vlog(stlog.debug, "Compacting {} adjacent segments:", segments.size());
+        vlog(gclog.debug, "Compacting {} adjacent segments:", segments.size());
         for (size_t i = 0; i < segments.size(); i++) {
-            vlog(stlog.debug, "Segment {}: {}", i + 1, *segments[i]);
+            vlog(gclog.debug, "Segment {}: {}", i + 1, *segments[i]);
         }
     }
```

Comment:
The log level check appears to be for the wrong logger.

Nit: Multi-line log is a bit awkward. .